core/Scene.js

```diff
@@ -122,20 +158,27 @@ Scene.prototype.onReceive = function onReceive (event, payload) {
     // and the context would receive its size the same way that any render size
     // component receives its size.
     if (event === 'CONTEXT_RESIZE') {
-        
-        if (payload.length < 2) 
+        if (payload.length < 2) {
             throw new Error(
-                    'CONTEXT_RESIZE\'s payload needs to be at least a pair' +
-                    ' of pixel sizes'
+                'CONTEXT_RESIZE\'s payload needs to be at least a pair' +
+                ' of pixel sizes'
             );
+        }
+
+        this._initialized = true;
+
+        while (this._messages.length)
+            this._updater.message(this._messages.shift());
 
         this.setSizeMode(Size.ABSOLUTE, Size.ABSOLUTE, Size.ABSOLUTE);
         this.setAbsoluteSize(payload[0],
                              payload[1],
                              payload[2] ? payload[2] : 0);
 
+
+         while (this._requesters.length)
+             this._requesters.shift().onUpdate();
```

Comment:
This would require the DOMElement, Mesh and Light to be rewritten in order to better account for the case of unmounted nodes.